### ISSUE TITLE
[Question] plan on supporting surreal 2.0?

### ISSUE THREAD

**Issue by AlexandreRoba** [Oyelowo/surreal-orm#74]
Hey @Oyelowo , great work on this orm. I've started playing with it and found it very promising. 

Do you have any plan on supporting Surreal 2.0?

**Comment by Oyelowo** [comment]
appreciate that @AlexandreRoba !
Glad you've found this useful.

I am on the last feature after which I plan to support Surreal 2.0.

Do you have plans to use this or just exploring?

**Comment by AlexandreRoba** [comment]
@Oyelowo Sorry I did not get the notification of the message. Yes we do have plan to use it. We are currently writing all our Data access by hand and it is becoming really difficult. I'll be glad to share with you were we are. Let me know if you want more information.

**Comment by Oyelowo** [comment]
@AlexandreRoba No worries!
Sounds good, I'm happy to hear more from you. Also, feel free to reach my email via oyelowo.oss@gmail.com if need be

**Comment by sawa-ko** [comment]
I love this ORM, I find it very versatile and functional. I plan to use it for my next project, but I looked at this issue and it says it doesn't support version 2.0, is it planned to support this version?

**Comment by laciferin2024** [comment]
Same , i m not sure whether i should opt for surreal db or sqlite with very limited support.

**Comment by Oyelowo** [comment]
Resumed work on this. Doing the preparation for the public release.

Plan to is to 
- release for 1.+ first,
- Set things up for the support of Postgres, Sqllie and MySQL and potentially mongodb
- then 2.0+ will follow. Already investigated the key changes necessary for v2.0

**Comment by sawa-ko** [comment]
I don't know, I feel that giving support for more SQL servers, is very overkill, I would like this ORM to focus only on surrealdb so that all the effort can be fully leveraged for surrealdb. But of course, you can do whatever you want with your library, it's just a personal opinion. I will always use your library anyway.

**Comment by Oyelowo** [comment]
@sawa-ko 
Fair enough, why do you feel that supporting more databases is overkill or unnecessary?

**Comment by laciferin2024** [comment]
No its not its the prisma way to do it. See IME, surreal doesn't support mobile platforms, so ORM makes it easier for us to migrate

**Comment by sawa-ko** [comment]
> @sawa-ko 
> Fair enough, why do you feel that supporting more databases is overkill or unnecessary?

In my opinion, I would say unnecessary.

**Comment by laciferin2024** [comment]
Yeah, i guess u haven't heard of success of `prisma`,`gorm` , `ActiveRecord` , `bun orm`.

**Comment by mariemachtmiau** [comment]
> > [@sawa-ko](https://github.com/sawa-ko)
> > Fair enough, why do you feel that supporting more databases is overkill or unnecessary?
> 
> In my opinion, I would say unnecessary.

Agreed, it feels like SurrealDB is special enough that trying to match the features with other DBs will be a lot of work which could go into making the SurrealDB integration itself better.
I personally would rather just have a really nice SurrealDB ORM which supports all of the features than another SeaORM for example

**Comment by d3v2a** [comment]
Great work!

Like many people here, I think integrating other databases into the project might be premature, since we’re already well supplied (Diesel, SeaORM, Toasty). 
It would probably be better to start with SurrealDB and then add support for other databases later to become a more general-purpose ORM.

> then 2.0+ will follow. Already investigated the key changes necessary for v2.0

Do you think it will require a lot of work?

Maybe we could help!
